EventFilter/GEMRawToDigi/plugins/GEMDigiToRawModule.cc

```diff
@@ -67,37 +69,63 @@ void GEMDigiToRawModule::produce(edm::StreamID iID, edm::Event& iEvent, edm::Eve
   std::vector<std::unique_ptr<AMC13Event>> amc13Events;
   amc13Events.reserve(FEDNumbering::MAXGEMFEDID - FEDNumbering::MINGEMFEDID + 1);
 
+  uint32_t LV1_id = iEvent.id().event();
+  uint16_t BX_id = iEvent.bunchCrossing();
+  uint32_t OrN = iEvent.orbitNumber();
+
+  // making map of bx GEMDigiCollection
+  // each bx will be saved as new AMC13Event, so GEMDigiCollection needs to be split into bx
+  std::map<int, GEMDigiCollection> gemBxMap;
+  for (auto range_iter = gemDigis->begin(); range_iter != gemDigis->end(); range_iter++) {
```

Comment:
This could be conveniently made a range-based for loop

Reply:
and done